PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp

```diff
@@ -1644,6 +1644,17 @@ bool FindReplaceDlg::processReplace(const TCHAR *txt2find, const TCHAR *txt2repl
 				msg += moreMatches?TEXT("The next occurence found"):TEXT("The next occurence not found");
 				setStatusbarMessage(msg, FSMessage);
 			}
+
+			if (pTextFind)
+			{
+				delete[] pTextFind;
+				pTextFind = nullptr;
+			}
+			if (pTextReplace)
+			{
+				delete[] pTextReplace;
+				pTextReplace = nullptr;
+			}

```

Comment:
Hmm. Wouldn't it be better to make this variable specific for "extended replace" branch of code.  So scope and purpose of it is much more clear. Something like:
```
if (isRegExp)
{
	replacedLen = (*_ppEditView)->replaceTargetRegExMode(txt2replace);
}
else
{
	if (replaceOptions._searchType == FindExtended)
	{
		int stringSizeReplace = lstrlen(txt2replace);
		TCHAR *txt2ReplaceExtended = new TCHAR[stringSizeReplace + 1];
		Searching::convertExtendedToString(txt2replace, txt2ReplaceExtended, stringSizeReplace);
 
		replacedLen = (*_ppEditView)->replaceTarget(txt2ReplaceExtended);
		
		delete[] txt2ReplaceExtended;
	}
	else
	{
		replacedLen = (*_ppEditView)->replaceTarget(txt2replace);
	}
}

```